PyInstaller/building/api.py

```diff
@@ -106,7 +106,11 @@ def __init__(self, *tocs, **kwargs):
         self.toc = []
         self.code_dict = {}
         for toc in tocs:
-            self.code_dict.update(getattr(toc, '_code_cache', {}))
+            # Check if code cache association exists for the given TOC list
+            code_cache = CONF['code_cache'].get(id(toc))
```

Comment:
Yeah, code_cache dicts are mappings of module names to their code objects, retrieved from modulegraph's nodes:
https://github.com/pyinstaller/pyinstaller/blob/fb545dba6d0fa7074bc32729734a9862caa8cc9a/PyInstaller/depend/analysis.py#L523-L542

Previously, they were attached to `pure` TOC objects and then collected in `PYZ` - although in practice, `PYZ` usually receives only one `pure` TOC, so it ended up with one such dictionary.

I tried to match the original behavior mostly because of possible corner cases - for example, spec instantiates two `Analysis` objects, with different search paths. In that case, I assume we might end up with different but eponymous modules, and would need separate code caches for each resulting PYZ and executable. 

It's not a very likely scenario (hopefully?), but I didn't want to delve too much into it at this point. I think we'll need to rework the part with code caches again when we switch from the old modulegraph, anyway.

Reply:
Fair enough. Let's leave it as is for now.